polkadot/zombienet-sdk-tests/tests/disabling/slashing.rs

```diff
@@ -0,0 +1,175 @@
+//! making some of the honest nodes go offline.
+
+use anyhow::anyhow;
+
+use crate::helpers::{assert_blocks_are_being_finalized, assert_para_throughput};
+use polkadot_primitives::{BlockNumber, CandidateHash, DisputeState, Id as ParaId, SessionIndex};
+use serde_json::json;
+use subxt::{OnlineClient, PolkadotConfig};
+use tokio::time::Duration;
+use tokio_util::time::FutureExt;
+use zombienet_sdk::NetworkConfigBuilder;
+
+#[tokio::test(flavor = "multi_thread")]
+async fn dispute_past_session_slashing() -> Result<(), anyhow::Error> {
+	let _ = env_logger::try_init_from_env(
+		env_logger::Env::default().filter_or(env_logger::DEFAULT_FILTER_ENV, "info"),
+	);
+
+	let images = zombienet_sdk::environment::get_images_from_env();
+
+	let config = NetworkConfigBuilder::new()
+		.with_relaychain(|r| {
+			r.with_chain("westend-local")
+				.with_default_command("polkadot")
+				.with_default_image(images.polkadot.as_str())
+				.with_default_args(vec![
+					"--no-hardware-benchmarks".into(),
+					"-lparachain=debug,runtime=debug".into(),
+				])
+				.with_genesis_overrides(json!({
+					"patch": {
+						"configuration": {
+							"config": {
+								"scheduler_params": {
+									"group_rotation_frequency": 3,
+									"max_validators_per_core": 1,
+									"lookahead": 3,
+								},
+								"needed_approvals": 2
+							}
+						}
+					}
+				}))
+				.with_node(|node| node.with_name("honest-validator-0"))
+				.with_node(|node| node.with_name("honest-validator-1"))
+				.with_node(|node| node.with_name("honest-flaky-validator-0"))
+				.with_node(|node| {
+					node.with_name("malicious-backer")
+						.with_image(
+							std::env::var("MALUS_IMAGE")
+								.unwrap_or("docker.io/paritypr/malus".to_string())
+								.as_str(),
+						)
+						.with_command("malus")
+						.with_subcommand("suggest-garbage-candidate")
+						.with_args(vec![
+							"--no-hardware-benchmarks".into(),
+							"--insecure-validator-i-know-what-i-do".into(),
+							"-lMALUS=trace,parachain=debug".into(),
+						])
+				})
+		})
+		.with_parachain(|p| {
+			p.with_id(1337)
+				.with_default_command("polkadot-parachain")
+				.with_default_image(images.cumulus.as_str())
+				.with_default_args(vec![
+					"--experimental-use-slot-based".into(),
+					"-lparachain=debug".into(),
+				])
+				.with_collator(|n| n.with_name("collator-1337"))
+		})
+		.build()
+		.map_err(|e| {
+			let errs = e.into_iter().map(|e| e.to_string()).collect::<Vec<_>>().join(" ");
+			anyhow!("config errs: {errs}")
+		})?;
+
+	let spawn_fn = zombienet_sdk::environment::get_spawn_fn();
+	let network = spawn_fn(config).await?;
+
+	let malus = network.get_node("malicious-backer")?;
+	malus.pause().await?;
+
+	let honest = network.get_node("honest-validator-0")?;
+	let relay_client: OnlineClient<PolkadotConfig> = honest.wait_client().await?;
+
+	// Wait for some para blocks being produced
+	assert_para_throughput(&relay_client, 20, [(ParaId::from(1337), 10..50)].into_iter().collect())
```

Comment:
this is a pretty large range of candidates backed per relay chain block. is it needed?

Reply:
yes, we need to wait for a couple of session, otherwise the chances of hitting the debug panic via https://github.com/paritytech/polkadot-sdk/issues/633 are high

Reply:
I don't get how in 20 relay chain blocks we could have 50 backed candidates of a single parachain (it's not using elastic scaling). I was suggesting that we should have a tighter range here

Reply:
Asserting the number of produced blocks slots is not the primary purpose of this test. I'd like to avoid it being more flaky in CI than necessary, where some of the pods are unreasonably slots at times.

Reply:
Ah, sorry you were commenting about the upper bound, yeah, lowered it now.